pandas/indexes/base.py

```diff
@@ -1464,13 +1464,13 @@ def append(self, other):
         names = set([obj.name for obj in to_concat])
         name = None if len(names) > 1 else self.name
 
-        typs = _concat.get_dtype_kinds(to_concat)
-
-        if 'category' in typs:
+        if self.is_categorical():
             # if any of the to_concat is category
```

Comment:
then you should change this comment as this is no longer true (the Index must be a CI to return a CI) only.